[BUG] TypeError: unhashable type: 'list' when using OpenAPI 3.1 schemas with type: [string, "null"]

### Checklist

- [ ] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq/) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Schemathesis fails to load an OpenAPI 3.1 schema that uses a union type (e.g. type: [string, "null"]) with a format.

This is a valid JSON Schema / OpenAPI 3.1 construct, but Schemathesis raises a TypeError: unhashable type: 'list' during schema processing.

It appears Schemathesis assumes schema["type"] is always a string, but in OpenAPI 3.1 it can also be a list (per JSON Schema 2020-12).

As a result, any schema using this pattern (for example, nullable date-time strings) causes test collection to fail.

### To Reproduce

1. Running python test like this:
```python
...

schema = schemathesis.openapi.from_asgi("/openapi.json", app=app)

@schema.parametrize()
def test_api(case):
    case.call_and_validate()
```
With an OpenAPI specification like this:

```yaml
openapi: 3.1.0
info:
  title: Example schema for nullable date-time
  version: 1.0.0
paths: {}
components:
  schemas:
    Metadata:
      type: object
      description: Example showing date-time with null support
      properties:
        publicationDate:
          type: [string, "null"]
          format: date-time
          description: Publication timestamp (RFC 3339, e.g., 2025-10-16T15:34:00Z)
```

2. See error

```
test_api_contract.py:None (test_api_contract.py)
../.venv/lib/python3.12/site-packages/schemathesis/pytest/plugin.py:244: in collect
    items = [item for operation in self.schema.get_all_operations() for item in self._gen_items(operation)]
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/pytest/plugin.py:159: in _gen_items
    funcobj = create_test(
../.venv/lib/python3.12/site-packages/schemathesis/generation/hypothesis/builder.py:183: in create_test
    hypothesis_test = add_coverage(
../.venv/lib/python3.12/site-packages/schemathesis/generation/hypothesis/builder.py:336: in add_coverage
    for case in generate_coverage_cases(
../.venv/lib/python3.12/site-packages/schemathesis/generation/hypothesis/builder.py:374: in generate_coverage_cases
    for case in _iter_coverage_cases(
../.venv/lib/python3.12/site-packages/schemathesis/generation/hypothesis/builder.py:652: in _iter_coverage_cases
    next_value = next(iterator)
                 ^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:500: in cover_schema_iter
    yield from cover_schema_iter(ctx, schema, seen)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:540: in cover_schema_iter
    yield from _negative_properties(ctx, template, value)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:1129: in _negative_properties
    for value in cover_schema_iter(nctx, sub_schema):
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:644: in cover_schema_iter
    positive = [case.value for case in cover_schema_iter(ctx.with_positive(), schema["items"])]
                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:500: in cover_schema_iter
    yield from cover_schema_iter(ctx, schema, seen)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:521: in cover_schema_iter
    yield from _cover_positive_for_type(ctx, schema, None)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:429: in _cover_positive_for_type
    yield from cover_schema_iter(ctx, sub_schema)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:500: in cover_schema_iter
    yield from cover_schema_iter(ctx, schema, seen)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:524: in cover_schema_iter
    yield from _cover_positive_for_type(ctx, schema, ty)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:459: in _cover_positive_for_type
    yield from _positive_object(ctx, schema, cast(dict, template))
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:1091: in _positive_object
    for new in cover_schema_iter(ctx, sub_schema):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:500: in cover_schema_iter
    yield from cover_schema_iter(ctx, schema, seen)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:524: in cover_schema_iter
    yield from _cover_positive_for_type(ctx, schema, ty)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:459: in _cover_positive_for_type
    yield from _positive_object(ctx, schema, cast(dict, template))
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:1091: in _positive_object
    for new in cover_schema_iter(ctx, sub_schema):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:524: in cover_schema_iter
    yield from _cover_positive_for_type(ctx, schema, ty)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:453: in _cover_positive_for_type
    yield from _positive_string(ctx, schema)
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:852: in _positive_string
    value = ctx.generate_from_schema(schema)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../.venv/lib/python3.12/site-packages/schemathesis/generation/coverage.py:276: in generate_from_schema
    return cached_draw(STRATEGIES_FOR_TYPE[schema["type"]])
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unhashable type: 'list'

During handling of the above exception, another exception occurred:
../.venv/lib/python3.12/site-packages/schemathesis/pytest/plugin.py:249: in collect
    pytest.fail("Error during collection")
E   Failed: Error during collection
```

### Expected behavior

My test succeeds.

### Environment

```
- OS: linux
- Python version: 3.12
- Schemathesis version: 4.3.9
- Spec version: Open API 3.1.0
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 ### :bug: Fixed
 
 - Internal error when generating invalid type in the coverage phase.
+- Internal error when generating valid instance against an array of types in the coverage phase. [#3194](https://github.com/schemathesis/schemathesis/issues/3194)
 
 ### :wrench: Changed
 

--- a/src/schemathesis/generation/coverage.py
+++ b/src/schemathesis/generation/coverage.py
@@ -8,6 +8,7 @@ from functools import lru_cache, partial
 from itertools import combinations
 
 from schemathesis.core.jsonschema.bundler import BUNDLE_STORAGE_KEY
+from schemathesis.generation.hypothesis.strategies import combine
 
 try:
     from json.encoder import _make_iterencode  # type: ignore[attr-defined]
@@ -74,6 +75,14 @@ STRATEGIES_FOR_TYPE = {
     "array": ARRAY_STRATEGY,
     "object": OBJECT_STRATEGY,
 }
+
+
+def get_strategy_for_type(ty: str | list[str]) -> st.SearchStrategy:
+    if isinstance(ty, str):
+        return STRATEGIES_FOR_TYPE[ty]
+    return combine([STRATEGIES_FOR_TYPE[t] for t in ty if t in STRATEGIES_FOR_TYPE])
+
+
 FORMAT_STRATEGIES = {**BUILT_IN_STRING_FORMATS, **get_default_format_strategies(), **STRING_FORMATS}
 
 UNKNOWN_PROPERTY_KEY = "x-schemathesis-unknown-property"
@@ -269,11 +278,11 @@ class CoverageContext:
         if isinstance(schema, bool):
             return 0
         keys = sorted([k for k in schema if not k.startswith("x-") and k not in ["description", "example", "examples"]])
-        if keys == ["type"] and isinstance(schema["type"], str) and schema["type"] in STRATEGIES_FOR_TYPE:
-            return cached_draw(STRATEGIES_FOR_TYPE[schema["type"]])
+        if keys == ["type"]:
+            return cached_draw(get_strategy_for_type(schema["type"]))
         if keys == ["format", "type"]:
             if schema["type"] != "string":
-                return cached_draw(STRATEGIES_FOR_TYPE[schema["type"]])
+                return cached_draw(get_strategy_for_type(schema["type"]))
             elif schema["format"] in FORMAT_STRATEGIES:
                 return cached_draw(FORMAT_STRATEGIES[schema["format"]])
         if (keys == ["maxLength", "minLength", "type"] or keys == ["maxLength", "type"]) and schema["type"] == "string":

--- a/test/coverage/test_combinations.py
+++ b/test/coverage/test_combinations.py
@@ -1102,6 +1102,11 @@ def test_negative_multiple_types(nctx):
     assert not cover_schema(nctx, schema)
 
 
+def test_positive_multiple_types(pctx):
+    schema = {"type": ["string", "null"], "format": "date-time"}
+    assert cover_schema(pctx, schema) == ["", None]
+
+
 @pytest.mark.parametrize(
     ("schema", "expected"),
     [

